### PR TITLE
[C++][Gandiva] Implement Translate Function

### PR DESCRIPTION
**Pull request by ViniciusSouzaRoque**

### CODE REVIEW

File: cpp/src/gandiva/gdv_function_stubs.cc [hunk 1]
```diff
@@ -1120,6 +1120,89 @@ const char* gdv_mask_last_n_utf8_int32(int64_t context, const char* data,
...
   return out;
 }
+
+GANDIVA_EXPORT
+const char* translate_utf8_utf8_utf8(int64_t context, const char* in, int32_t in_len,
+                                     const char* from, int32_t from_len, const char* to,
+                                     int32_t to_len, int32_t* out_len) {
+  if (in_len <= 0) {
+    *out_len = 0;
+    return nullptr;
```
Comment (anthonylouisbsb) [inline cpp/src/gandiva/gdv_function_stubs.cc:1130]: returns an empty string `""`

File: cpp/src/gandiva/gdv_function_stubs.cc [hunk 2]
```diff
@@ -1120,6 +1120,89 @@ const char* gdv_mask_last_n_utf8_int32(int64_t context, const char* data,
...
+    return in;
+  }
+
+  // This variable is for receive the substitutions
+  char* result = reinterpret_cast<char*>(gdv_fn_context_arena_malloc(context, in_len));
+
+  if (result == nullptr) {
+    gdv_fn_context_set_error_msg(context, "Could not allocate memory for output string");
+    *out_len = 0;
+    return nullptr;
```
Comment (anthonylouisbsb) [inline cpp/src/gandiva/gdv_function_stubs.cc:1144]: ditto

File: cpp/src/gandiva/gdv_function_stubs.cc [hunk 3]
```diff
@@ -1120,6 +1120,89 @@ const char* gdv_mask_last_n_utf8_int32(int64_t context, const char* data,
...
+  const char empty = '\0';
+
+  for (int in_for = 0; in_for < in_len; in_for++) {
+    if (subs_list.find(in[in_for]) != subs_list.end()) {
+      if (subs_list[in[in_for]] != empty) {
+        // If exist in map, only add the correspondent value in result
+        result[result_len] = subs_list[in[in_for]];
+        result_len++;
+      }
+      continue;
```
Comment (anthonylouisbsb) [inline cpp/src/gandiva/gdv_function_stubs.cc:1172]: is this `continue` necessary?

**Review by anthonylouisbsb — CHANGES_REQUESTED** [review]

File: cpp/src/gandiva/gdv_function_stubs.cc (hunk 1 above)
Comment (ViniciusSouzaRoque) [reply cpp/src/gandiva/gdv_function_stubs.cc]: Done

File: cpp/src/gandiva/gdv_function_stubs.cc (hunk 2 above)
Comment (ViniciusSouzaRoque) [reply cpp/src/gandiva/gdv_function_stubs.cc]: Done

File: cpp/src/gandiva/gdv_function_stubs.cc (hunk 3 above)
Comment (ViniciusSouzaRoque) [reply cpp/src/gandiva/gdv_function_stubs.cc]: no, i removed. 
done

**Pull request closed without merging**